caas/kubernetes/provider/k8s.go

```diff
@@ -124,34 +132,59 @@ func NewK8sBroker(
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+
 	modelUUID := newCfg.UUID()
-	namespace := newCfg.Name()
-	if namespace == environsbootstrap.ControllerModelName {
-		// namespace format: <controller>-<controller-model-UUID> for juju controller to achieve:
-		// 1. multi controllers running in same k8s cluster;
-		// 2. avoid potential conflict if there is existing namespace named "controller"(warning:
-		//    an non juju related existing namespace could be destroyed if bootstrapping failed due
-		//    to AlreadyExistedNamespace error);
-
-		// IMPORTANT!
-		// TODO(bootstrap): do we want to run multi controller in same k8s cluster or always run one?
-		// Solution(potential): we SHOULD label all namespaces(controller ns, model ns) with controller ID,
-		// and always check if the controller owns(created) the namespace whenever it tries to
-		// CRUD any resources inside the namespace.
-		// Note: we should consider above `Solution` even we always run single controller per cluster, because
-		// we should NEVER allow juju controller to touch any namespace that was NOT created by JUJU.
-		namespace += "-" + modelUUID[:8]
-		logger.Debugf("found config name %q, so naming namespace to %q", newCfg.Name(), namespace)
-	}
-	return &kubernetesClient{
+	if modelUUID == "" {
+		return nil, errors.NotValidf("modelUUID is required")
+	}
+	client := &kubernetesClient{
 		clock:               clock,
 		Interface:           k8sClient,
 		apiextensionsClient: apiextensionsClient,
-		namespace:           namespace,
 		envCfg:              newCfg.Config,
+		namespace:           newCfg.Name(),
```

Comment:
should show know the final namespace here and not have to set it again using setNamespace()